src/lib/core/utils/data-element-constraints.ts

```diff
@@ -218,7 +218,9 @@ function variableConstraintPredicate(
   );
 }
 
-export type VariablesByInputName = Partial<Record<string, VariableDescriptor>>;
+export type VariablesByInputName<T extends string = string> = Partial<
```

Comment:
**comment**: TypeScript n00b here! What is `T extends string = string>` doing?

Reply:
And the `= string` is a fall-back if the "user" doesn't use the generic-style declaration.

Sorry I never did CS so I don't use the correct words all the time.

Reply:
What @bobular said is correct. Another way to say it is, the type parameter `T` must extend `string`, and will default to `string` if not specified by the consumer.